src/module/RegisterMappingFactory.cpp

```diff
@@ -31,54 +31,176 @@ std::shared_ptr<RegisterMapping> RegisterMappingFactory::fromJSONMapping(const M
+        std::transform(lowerCase.cbegin(), lowerCase.cend(), lowerCase.begin(), ::tolower);
+        bool* defaultValue = nullptr;
+        if (lowerCase == "true")
+            defaultValue = new bool(true);
+        else if (lowerCase == "false")
+            defaultValue = new bool(false);
+
+        // Make the mapping based on the operation
         if (jsonMapping.getOperationType() == RegisterMapping::OperationType::TAKE_BIT)
         {
-            return std::make_shared<BoolMapping>(jsonMapping.getReference(), jsonMapping.getRegisterType(),
-                                                 jsonMapping.getAddress(), jsonMapping.getOperationType(),
-                                                 jsonMapping.getBitIndex(), jsonMapping.isReadRestricted(), -1,
-                                                 jsonMapping.getFrequencyFilterValue());
+            mapping = std::make_shared<BoolMapping>(
+              jsonMapping.getReference(), jsonMapping.getRegisterType(), jsonMapping.getAddress(),
+              jsonMapping.getOperationType(), jsonMapping.getBitIndex(), jsonMapping.isReadRestricted(), -1,
+              jsonMapping.getFrequencyFilterValue(), std::chrono::milliseconds(jsonMapping.getRepeat()), defaultValue);
         }
-        return std::make_shared<BoolMapping>(jsonMapping.getReference(), jsonMapping.getRegisterType(),
-                                             jsonMapping.getAddress(), jsonMapping.isReadRestricted(), -1,
-                                             jsonMapping.getFrequencyFilterValue());
+        else
+        {
+            mapping = std::make_shared<BoolMapping>(jsonMapping.getReference(), jsonMapping.getRegisterType(),
+                                                    jsonMapping.getAddress(), jsonMapping.isReadRestricted(), -1,
+                                                    jsonMapping.getFrequencyFilterValue(),
+                                                    std::chrono::milliseconds(jsonMapping.getRepeat()), defaultValue);
+        }
+
+        // Free the default value if necessary
+        delete defaultValue;
+        return mapping;
+    }
     case RegisterMapping::OutputType::UINT16:
-        return std::make_shared<UInt16Mapping>(jsonMapping.getReference(), jsonMapping.getRegisterType(),
-                                               jsonMapping.getAddress(), jsonMapping.isReadRestricted(), -1,
-                                               jsonMapping.getDeadbandValue(), jsonMapping.getFrequencyFilterValue());
+    {
+        // Make place for the mapping
+        auto mapping = std::shared_ptr<UInt16Mapping>{};
+
+        // Check if the default value is anything
+        const auto& stringValue = jsonMapping.getDefaultValue();
+        std::uint16_t* defaultValue = nullptr;
+        try
+        {
+            auto value = std::stoul(stringValue);
+            defaultValue = new std::uint16_t(static_cast<std::uint16_t>(value));
```

Comment:
Same here. You already forgot to call delete. Just use address of local variable or pass null explicitly if default does not exist